java/vector/src/test/java/org/apache/arrow/vector/TestValueVector.java

```diff
@@ -2245,4 +2253,217 @@ public void testGetNullFromVariableWidthVector() {
       assertNull(varBinaryVector.get(0));
     }
   }
+
+  @Test
+  public void testIntVectorEquals() {
```

Comment:
can you please add some tests to verify that the validity bits are being checked too.

Reply:
Sure, done.